### PR TITLE
Move mypy configs from pre-commit to mypy config

### PR DESCRIPTION
**Pull request by max-sixty**
- [x] Passes `pre-commit run --all-files`

This should mean that running `mypy .` doesn't operate differently from running it in pre-commit.

It's also possible I've misunderstood something about mypy configs, please let me know if so!

### CODE REVIEW

**Comment by max-sixty** [comment]
This doesn't seem to work: 

```
conftest.py: error: Duplicate module named 'conftest' (also at './properties/conftest.py')
conftest.py: note: Are you missing an __init__.py? Alternatively, consider using --exclude to avoid checking one of them.
Found 1 error in 1 file (errors prevented further checking)
```

Not sure exactly why it doesn't ignore the errors given the configs.

I'll close this unless anyone has ideas.

**Comment by keewis** [comment]
in order to get this to work you would need to add empty `__init__.py` to `asv_bench` and `properties`. Not sure if we would then need to modify the packaging configuration, though.

**Comment by shoyer** [comment]
Maybe the recommended command should be `mypy xarray`?

**Comment by max-sixty** [comment]
I think this should now work! TIL toml doesn't have quotations for strings

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ repos:
     rev: v0.812
     hooks:
       - id: mypy
+        # Copied from setup.cfg
         exclude: "properties|asv_bench"
```
Comment (andersy005) [inline .pre-commit-config.yaml:38]: Now that `setup.cfg` has 

```
[mypy]
exclude = properties|asv_bench|doc
```

do we still need this `exclude` option here?

Comment (max-sixty) [reply .pre-commit-config.yaml:38]: We do, unfortunately.

I guess that pre-commit gives it a list of files, and when passed a list of files mypy doesn't honor the exclusions.

**Comment by keewis** [comment]
could you change the call in the mypy CI to use `python -m mypy .` (so we can verify this works)?

**Comment by max-sixty** [comment]
> could you change the call in the mypy CI to use `python -m mypy .` (so we can verify this works)?

Great done. 

I didn't know about that. Why do we have that as well as `ci-pre-commit.yaml`?

**Comment by keewis** [comment]
> Why do we have that as well as `ci-pre-commit.yaml`?

`pre-commit` does not create a full environment before running `mypy`, which means that it can't check using external type hints (like the ones from `numpy`). See #4881 for more explanation.

**Comment by max-sixty** [comment]
> `pre-commit` does not create a full environment before running `mypy`, which means that it can't check using external type hints (like the ones from `numpy`). See #4881 for more explanation.

Thanks! I added a link to that for future travelers

**Pull request merged**